### PR TITLE
Change seamless panels to inherit any parent's colours

### PR DESCRIPTION
**Pull request by jingting1412**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [x] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

**Overview of changes:**
#1697 
Changed the seamless panels so that the background is transparent and it inherits the colours of any parents its contained in. 

**Anything you'd like to highlight/discuss:**
As pointed out in the issue, this might break some edge cases (e.g. a seamless panel in a `<box background-color="black" color="black">` box). 

**Testing instructions:**
I've added some examples in the userguide to show the new look. `markbind serve -d` the docs to see the changes and try out different cases to see if it works or if there's any improvements.

**Proposed commit message: (wrap lines at 72 characters)**
Change seamless panels to inherit colours of any parents

---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No unrelated changes 

---

**Reviewer checklist:**

Indicate the [SEMVER](https://semver.org/) impact of the PR:
- [ ] Major (when you make incompatible API changes)
- [ ] Minor (when you add functionality in a backward compatible manner)
- [x] Patch (when you make backward compatible bug fixes)

At the end of the review, please label the PR with the appropriate label: `r.Major`, `r.Minor`, `r.Patch`.

Breaking change release note preparation (if applicable):
- To be included in the release note for any feature that is made obsolete/breaking

> Give a brief explanation note about:
>   - what was the old feature that was made obsolete
>   - any replacement feature (if any), and
>   - how the author should modify his website to migrate from the old feature to the replacement feature (if possible).

### CODE REVIEW

**Comment by jingting1412** [comment]
Idk how my other allocate space for scrollbar commit got here 😞 but the changed files only reflect the changes relevant to this pr!

**Comment by kaixin-hc** [comment]
Weird that the netlify deploy preview failed. Maybe it will fix itself on push

File: docs/userGuide/syntax/panels.md [hunk 1]
```diff
@@ -104,6 +104,29 @@ plain text ...
 </variable>
 </include>
 
+<box>
```
Comment (yucheng11122017) [inline docs/userGuide/syntax/panels.md:107]: TBH i think this is too much documentation and examples.. I would rather just keep one, preferably the first because the other two are very blinding and I am old. 

In fact... I'm a bit tempted to not have documentation for this because I feel like its too small and specific feature. What do the rest of the reviewers think?

File: packages/vue-components/src/panels/NestedPanel.vue [hunk 2]
```diff
@@ -324,6 +346,17 @@ export default {
         align-items: center;
     }
 
+    .morph-display-wrapper-seamless {
+        margin-top: 5px;
+        display: flex;
+        align-items: center;
+        color: inherit;
```
Comment (yucheng11122017) [inline packages/vue-components/src/panels/NestedPanel.vue:353]: Is the difference between `.morph-display-wrapper-seamless` and `.morph-display-wrapper` only the inheritance of color? 
In that case, could we make it such that `.morph-display-wrapper-seamless` is just `color: inherit`?
So the button will always have the class `.morph-display-wrapper` and will have the class `.morph-display-wrapper-seamless`  if it is seamless

File: packages/vue-components/src/panels/NestedPanel.vue [hunk 3]
```diff
@@ -27,9 +32,9 @@
     ref="cardContainer"
     :class="['card-container', addClass]"
   >
-    <div :class="['card', { 'expandable-card': isExpandableCard }, borderType]">
+    <div :class="['card', { 'expandable-card': isExpandableCard, 'card-seamless': isSeamless }, borderType]">
       <div
-        :class="['card-header',{'header-toggle':isExpandableCard}, cardType, borderType]"
+        :class="['card-header', {'header-toggle':isExpandableCard}, cardType, borderType]"
```
Comment (yucheng11122017) [inline packages/vue-components/src/panels/NestedPanel.vue:37]: Unnecessary change

File: packages/vue-components/src/panels/NestedPanel.vue [hunk 4]
```diff
@@ -151,12 +159,13 @@ export default {
...
       if (this.isSeamless) {
-        return 'bg-white';
+        return 'bg-transparent';
       }
       return `bg-${this.type || 'light'}`;
     },
     isLightBg() {
-      return this.cardType === 'bg-light' || this.cardType === 'bg-white' || this.cardType === 'bg-warning';
+      return this.cardType === 'bg-light'
+      || this.cardType === 'bg-warning';
```
Comment (yucheng11122017) [inline packages/vue-components/src/panels/NestedPanel.vue:168]: Can we make this one line?

**Review by yucheng11122017 — COMMENTED** [review]
Hi @jingting1412 thanks for the work! 
Functionality and all looks great - just some minor nits. 

Could you also add some test cases for this behaviour? I think you should be able to add unit tests without too much issue

File: docs/userGuide/syntax/panels.md (hunk 1 above)
Comment (kaixin-hc) [reply docs/userGuide/syntax/panels.md:107]: Just one is probably good enough! But the first one is not obvious enough - maybe play with the colours a bit so it stands out more.
Screenshot 2024-04-06 at 8 11 17 PM

I'm team documentation here, the use case becomes far clearer when you see how the seamless panel looks inside a colored background.

File: packages/vue-components/src/panels/NestedPanel.vue [hunk 5]
```diff
@@ -279,6 +299,8 @@ export default {
 
     .card-collapse > hr {
         margin-top: 0;
+        background-color: currentcolor;
+        opacity: 0.5;
```
Comment (kaixin-hc) [inline packages/vue-components/src/panels/NestedPanel.vue:303]: a bit confused what does this affect!

File: docs/userGuide/syntax/panels.md (hunk 1 above)
Comment (jingting1412) [reply docs/userGuide/syntax/panels.md]: Yep I just left one example on the user guide so that user is at least aware of this property of seamless panels. Please tell me if the colour combo is not nice 😅

File: packages/vue-components/src/panels/NestedPanel.vue (hunk 2 above)
Comment (jingting1412) [reply packages/vue-components/src/panels/NestedPanel.vue:349]: Yep sorry about that, I have made the changes accordingly

File: packages/vue-components/src/panels/NestedPanel.vue (hunk 3 above)
Comment (jingting1412) [reply packages/vue-components/src/panels/NestedPanel.vue]: Sorry about that, have made the changes accordingly

File: packages/vue-components/src/panels/NestedPanel.vue (hunk 4 above)
Comment (jingting1412) [reply packages/vue-components/src/panels/NestedPanel.vue]: Yep sure! I've changed it to one line

File: packages/vue-components/src/panels/NestedPanel.vue (hunk 5 above)
Comment (jingting1412) [reply packages/vue-components/src/panels/NestedPanel.vue:302]: This is to ensure that the page break also inherits the colour of the text, or else it would always be a dark colour. 
For example with the inherit property:
Screenshot 2024-04-07 at 4 55 42 PM

without the inherit property:
Screenshot 2024-04-07 at 4 55 21 PM

As you can see without these changes the page break might sometimes be unnoticeable against the parent's colour

File: docs/userGuide/syntax/panels.md [hunk 6]
```diff
@@ -104,6 +104,14 @@ plain text ...
 </variable>
 </include>
 
+<box background-color="#C51E3A" color="white">
+  
+  :bulb: Seamless panels inherit the background colour and text colour of any parents!
+  <br/>
+  <panel type="seamless" header="This is an example seamless panel" popup-url="/userGuide/syntax/extra/loadContent.html">
```
Comment (kaixin-hc) [inline docs/userGuide/syntax/panels.md:111]: Why the popup-URL ?

File: docs/userGuide/syntax/panels.md (hunk 1 above)
Comment (kaixin-hc) [reply docs/userGuide/syntax/panels.md]: Colour looks good to me : )

File: packages/vue-components/src/panels/NestedPanel.vue [hunk 7]
```diff
@@ -5,7 +5,12 @@
     :class="['card-container', addClass]"
   >
     <span class="morph">
-      <button :class="['morph-display-wrapper', 'btn', btnType]" @click="open()">
+      <button
+        :class="[isSeamless ? 'morph-display-wrapper-seamless': 'morph-display-wrapper',
+                 'btn',
```
Comment (yucheng11122017) [inline packages/vue-components/src/panels/NestedPanel.vue:10]: Shouldn't this button has the `morph-display-wrapper` anyway regardless of whether it is seamless or not?

**Review by yucheng11122017 — COMMENTED** [review]
Small nit!

File: packages/vue-components/src/panels/NestedPanel.vue (hunk 7 above)
Comment (jingting1412) [reply packages/vue-components/src/panels/NestedPanel.vue:10]: I added the new `morph-display-wrapper-seamless` since I've noticed that the colour of the minimized panels would not inherit the parent's colour with just `morph-display-wrapper`

With just `morph-display-wrapper`:

https://github.com/MarkBind/markbind/assets/105090139/3bf36d36-47ae-40d8-a6fa-5054a9eef68c

With `morph-display-wrapper-seamless` and `morph-display-wrapper`

https://github.com/MarkBind/markbind/assets/105090139/ef8dbacb-5b01-49ff-ad9c-e8dc54e25af7

File: docs/userGuide/syntax/panels.md (hunk 6 above)
Comment (jingting1412) [reply docs/userGuide/syntax/panels.md:111]: I'm just making sure that all buttons follow the seamless behaviour when testing, I'll remove from the final example!

File: packages/vue-components/src/panels/NestedPanel.vue (hunk 7 above)
Comment (yucheng11122017) [reply packages/vue-components/src/panels/NestedPanel.vue:10]: As in does the a seamless button also need the properties from `morph-display-wrapper`?

**Review by yucheng11122017 — APPROVED** [review]
LGTM except for one small clarification!

File: packages/vue-components/src/panels/NestedPanel.vue (hunk 7 above)
Comment (jingting1412) [reply packages/vue-components/src/panels/NestedPanel.vue:10]: No seamless buttons wouldn't need properties from `morph-display-wrapper`, seamless buttons only appear when the panel is not minimised so it shouldn't need any properties from `morph-display-wrapper`

Comment (kaixin-hc) [reply packages/vue-components/src/panels/NestedPanel.vue:10]: I've taken a video for your reference that shows that toggling these properties would make a visual difference. The caret button seems to also be a seamless button which appears when the panel is minimised, so the non seamless panels and the seamless panels then have slightly different styling due to this line of code. 

https://github.com/MarkBind/markbind/assets/59471320/8be20b60-03e2-4759-89e2-481bb39f2094

File: packages/vue-components/src/panels/NestedPanel.vue (hunk 2 above)
Comment (jingting1412) [reply packages/vue-components/src/panels/NestedPanel.vue:349]: I've changed it now so that both the seamless and non-seamless morph wrapper would share the same basic styles, and only `morph-display-wrapper-seamless` will have the `color:inherit` property. This allows for future changes to the styles of all morph wrappers to be done in one place and allows the seamless wrapper to have the same styles as the other wrappers

File: packages/vue-components/src/panels/NestedPanel.vue (hunk 7 above)
Comment (yucheng11122017) [reply packages/vue-components/src/panels/NestedPanel.vue:10]: Rather than doing this or else statement, could all buttons have `morph-display-wrapper` by default and then only seamless buttons will have `morph-display-wrapper-seamless`?

Comment (jingting1412) [reply packages/vue-components/src/panels/NestedPanel.vue:11]: yep sure I've implemented it as such now

**Review by yucheng11122017 — APPROVED** [review]
LGTM!
